src/tests/ffi-key.cpp

```diff
@@ -4111,3 +4111,196 @@ dbpkwTCmBVbAmazgea0B
     assert_int_equal(keys, 1);
     rnp_ffi_destroy(ffi);
 }
+
+/* Functions below are used to demonstrate how to check whether key has weak MD5/SHA1
+ * signatures, and may be reused later in FFI code */
+static bool
+is_self_signature(const char *keyid, rnp_signature_handle_t sig)
+{
+    char *signer = NULL;
+    rnp_signature_get_keyid(sig, &signer);
+    if (!signer) {
+        return false;
+    }
+    bool result = !strcmp(keyid, signer);
+    rnp_buffer_destroy(signer);
+    return result;
+}
+
+static bool
+is_weak_signature(rnp_ffi_t ffi, rnp_signature_handle_t sig)
+{
+    char *   hash = NULL;
+    uint32_t creation = 0;
+    rnp_signature_get_hash_alg(sig, &hash);
+    rnp_signature_get_creation(sig, &creation);
+    /* This approach would be more general, however hardcoding MD5/SHA1 may be used as well */
+    uint32_t level = 0;
+    rnp_get_security_rule(ffi, RNP_FEATURE_HASH_ALG, hash, creation, NULL, NULL, &level);
+    bool res = level < RNP_SECURITY_DEFAULT;
+    if (res) {
+        printf(
+          "Detected weak signature with %s hash, created at %zu\n", hash, (size_t) creation);
+    }
+    rnp_buffer_destroy(hash);
+    return res;
+}
+
+const std::string
+get_uid_str(rnp_uid_handle_t uid)
+{
+    uint32_t type = 0;
+    rnp_uid_get_type(uid, &type);
+    switch (type) {
+    case RNP_USER_ID: {
+        void * data = NULL;
+        size_t len = 0;
+        rnp_uid_get_data(uid, &data, &len);
+        std::string res((const char *) data, (const char *) data + len);
+        rnp_buffer_destroy(data);
+        return res;
+    }
+    case RNP_USER_ATTR:
+        return "photo";
+    default:
+        return "Unknown";
+    }
+}
+
+static size_t
+key_weak_self_signatures_count(rnp_ffi_t ffi, rnp_key_handle_t key)
```

Comment:
Maybe `get_key_weak_self_signatures_count()` or `key_get_weak_self_signatures_count()`?

Reply:
@antonsviridenko Is it good to go or you still insist on the naming change?